lib/src/store.dart

```diff
@@ -146,4 +147,24 @@ class Store extends Stream<Store> with Disposable {
       trigger();
     }));
   }
+
+  /// A convenience method for listening to an [action] and triggering
+  /// automatically once the callback for said action has completed.
+  ///
+  /// If [onAction] is provided, it will be called every time [action] is
+  /// dispatched. If [onAction] returns a [Future], [trigger] will not be
+  /// called until that future has resolved.
+  ///
+  /// If the `Store` has been disposed, this method throws a [StateError].
+  onActionTrigger<T>(Action<T> action, [FutureOr<dynamic> onAction(T payload)]) {
```

Comment:
How's the name? How's FutureOr<dynamic> ? I changed it from `void` based on the documentation saying that the onAction method can return a Future.

Reply:
I would personally `void` and update the docs.